File: composite/java/Composite.java
package composite.java;

import java.util.ArrayList;
import java.util.List;

class Directory implements FileSystemComponent {
    private String name;
    private List<FileSystemComponent> components = new ArrayList<>();

    public Directory(String name) {
        this.name = name;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getSize() {
        return components.stream().mapToInt(FileSystemComponent::getSize).sum();
    }

    @Override
    public void add(FileSystemComponent component) {
        components.add(component);
    }

    @Override
    public void remove(FileSystemComponent component) {
        components.remove(component);
    }
}

File: composite/python/Composite.py
from Component import FileSystemComponent

class Directory(FileSystemComponent):
    def __init__(self, name):
        self.name = name
        self.components = []
        
    def get_name(self):
        return self.name()
    
    def get_size(self):
        return sum([component.get_size() for component in self.components])
    
    def add(self, component):
        self.components.append(component)
        
    def remove(self, component):
        self.components.remove(component)